lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -0,0 +1,745 @@
+        return self.send("PATCH", path, data, headers)
+
+    def update(self, path, data=None, headers=None):
+        return self.send("UPDATE", path, data, headers)
+
+
+def is_present(compute_api, server):
+    print_warning(compute_api, "We test to see if the server %s it's present or not" % server["id"])
+    return True
+
+
+def create_server(compute_api, server):
+    print_warning(compute_api, "Starting a create_server")
+    target_server = None
+    response = compute_api.post(path="servers",
+                                data={"enable_ipv6": server["enable_ipv6"],
+                                      "tags": server["tags"],
+                                      "commercial_type": server["commercial_type"],
+                                      "image": server["image"],
+                                      "name": server["name"],
+                                      "organization": server["organization"]})
+
+    if not response.ok:
+        msg = 'Error during server creation: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    try:
+        target_server = response.json["server"]
+    except KeyError:
+        compute_api.module.fail_json(msg="Error in getting the server information from: %s" % response.json)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(create_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=target_server)
+
+    return target_server
+
+
+def restart_server(compute_api, server):
+    print_warning(compute_api, "Starting restart_server")
+    response = compute_api.post(path="servers/%s/action" % server["id"],
+                                data={"action": "reboot"})
+
+    if not response.ok:
+        msg = 'Error during server reboot: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=msg)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(restart_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=server)
+
+    return response
+
+
+def stop_server(compute_api, server):
+    print_warning(compute_api, "Starting stop_server strategy")
+    response = compute_api.post(path="servers/%s/action" % server["id"],
+                                data={"action": "poweroff"})
+
+    print_warning(compute_api=compute_api, msg=response.json)
+    if not response.ok:
+        err_msg = 'Error during server power-off: (%s) %s' % (response.status_code, response.json)
+        compute_api.module.fail_json(msg=err_msg)
+
+    wait = compute_api.module.params["wait"]
+
+    if wait:
+        print_warning(compute_api, "(stop_server) Wait is enabled")
+        wait_to_complete_state_transition(compute_api=compute_api, server=server)
+
+    return response
+
+
+def start_server(compute_api, server):
```

Comment:
methods `start_server`/`stop_server`/`restart_server` are almost the same and could be factorized.